packages/synapse-react-client/src/components/CardDeck/TableQueryCardDeck.tsx

```diff
@@ -0,0 +1,137 @@
+import { CardDeck } from '@/components/CardDeck/CardDeck'
+import { CardDeckCardProps } from '@/components/CardDeck/CardDeckCardProps'
+import { ImageFileHandle } from '@/components/widgets/ImageFileHandle'
+import { useGetEntity } from '@/synapse-queries/entity/useEntity'
+import useGetQueryResultBundle from '@/synapse-queries/entity/useGetQueryResultBundle'
+import { SynapseConstants } from '@/utils'
+import { getFieldIndex } from '@/utils/functions/queryUtils'
+import { QueryBundleRequest } from '@sage-bionetworks/synapse-types'
+import { CardLink } from '../CardContainer/CardLink'
+import { getFileHandleAssociation, getLinkParams } from '../GenericCard'
+
+export type TableQueryCardDeckProps = {
+  entityId: string /* The table entity which should be queried */
+  titleColumnName: string /* The column name for the title */
+  descriptionColumnName: string /* The column name for the description */
+  ctaButtonTextColumnName: string /* The column name for the call to action button text */
+  ctaButtonURLColumnName: string /* The column name for the call to action button URL */
+  titleIconFileHandleColumnName?: string /* The column name for the title icon image file */
+  headerImageFileHandleColumnName?: string /* The column name for the header image file */
+  cardDeckType?: 'cckp' | 'b2ai' /* The type of card deck (cckp or b2ai)*/
+  linkConfig: CardLink /* The link configuration */
+}
+
+/**
+ * Transforms a Synapse table into a card deck.
+ */
+export function TableQueryCardDeck(props: TableQueryCardDeckProps) {
```

Comment:
Identical logic/props to the old `CardDeck`. In CCKP (and other portals where we want a Table to populate cards), we will use this